http/http.go

```diff
@@ -52,10 +52,29 @@ func Run(mux *http.ServeMux) {
 	}
 }
 
-func handleError(w http.ResponseWriter, r *http.Request, msg string, code int) {
-	http.Redirect(w, r, fmt.Sprintf("/fail?reason=%s", msg), code)
+func (s Server) handleHTMLError(w http.ResponseWriter, r *http.Request, msg string, code int, err error) {
+	// http.Error(w, fmt.Sprintf("failed: %s", msg), code)
+	w.WriteHeader(code)
 
-	log.Fatalf("Error code %v /n %s", code, msg)
+	err = s.Template.ExecuteTemplate(w, "fail.html.tmpl", msg)
+	if err != nil {
+		log.Fatalf("failed to execute %v", err)
+	}
+
+	fmt.Printf("failed: %s \n code %v \n %s", msg, code, err)
+}
+
+func handleJSONError(w http.ResponseWriter, r *http.Request, msg string, code int, err error) {
+	// http.Error(w, fmt.Sprintf("failed: %s", msg), code)
+	w.WriteHeader(code)
+	w.Header().Set("Content-Type", "application/json")
+
+	err = json.NewEncoder(w).Encode(msg)
```

Comment:
You're not using the `err` parameter here. I would create a struct to encode, like:
```go
data := struct{
  Message string
  Error string
} {
  Message: msg,
  Error: err.Error(),
}
err = json.NewEncoder(w).Encode(data)
```